Lessons_19/task1.py

```diff
@@ -0,0 +1,44 @@
+"""Task 1
+Create your own implementation of a built-in function enumerate,
+named 'with_index', which takes two parameters: 'iterable' and
+'start', default is 0.
+Tips: see the documentation for the enumerate function
+"""
+
+
+class MyEnumerate:
+    """Create custom enumerate object"""
+
+    def __init__(self, iterable, start=0):
+        self.index = 0
+        self.start = start
+        self.iterable = iterable or []
+
+    def __iter__(self):
+        self.index = 0
+        return self
+
+    def __next__(self):
+        if self.index == len(self.iterable):
+            raise StopIteration
+        res = (self.index + self.start, self.iterable[self.index])
+        self.index += 1
+        return res
+
+
+def with_index(iterable, start):
```

Comment:
You could have defined it as a simpler generator function with `yield`, but this solution will also do